src/core/net/dns_client.hpp

```diff
@@ -766,9 +786,21 @@ class Client : public InstanceLocator, private NonCopyable
     using RetryTimer = TimerMilliIn<Client, &Client::HandleTimer>;
 
     Ip6::Udp::Socket mSocket;
-    QueryList        mQueries;
-    RetryTimer       mTimer;
-    QueryConfig      mDefaultConfig;
+
+#if OPENTHREAD_CONFIG_DNS_OVER_TCP
+    Ip6::Tcp::Endpoint mEndpoint;
+
+    otLinkedBuffer mSendLink;
+    uint8_t        mSendBufferBytes[1024];
```

Comment:
Not sure if  we can somehow use OtMessage instead of raw buffers - but I guess it's TCP implementation decision. What about to define two new configs e.g. `OPENTHREAD_CONFIG_DNS_OVER_TCP_QUERY_MAX_SIZE`? We could then check if we don't overflow in the `Query` method.

Reply:
@abtink any thoughts about this part? I agree with @LuDuda that using OtMessage would be more convenient for developers using TCP. 

@LuDuda Configs are good Idea I will ad those and set defaults to 1024.

Reply:
@samkumar helped integrate TCLP into OT and implement the OT TCP APIs. There were good discussions on API design and current set of APIs were implemented (for good reasons). More details at https://github.com/openthread/openthread/pull/6491.

I agree that `otMessage` may be easier under certain situations (like here). At some point it may make sense to add a new set of parallel TCP API which use `otMessage`, and/or mechanism to convert an allocated `Message` into an `otLinkBuffer` (so to then use it with TCP API). I expect these would be bigger changes.

Reply:
@abtink thanks for info.